shapefile.py

```diff
@@ -1010,12 +1010,14 @@ def record(self, *recordList, **recordDict):
         # Balance if already not balanced
         if self.autoBalance and self.recNum > self.shpNum:
             self.balance()
-            
+
         record = []
         fieldCount = len(self.fields)
         # Compensate for deletion flag
         if self.fields[0][0].startswith("Deletion"): fieldCount -= 1
         if recordList:
+            if recordList[0] is tuple:
+                recordList = recordList[0]
             record = [recordList[i] for i in range(fieldCount)]

```

Comment:
Thanks @alexmnt. Could you perhaps expand on what you mean by adding a dynamic list and why this would be important to support, I'm not sure I understand? 

The code you changed simply checks if the first item in recordList is a tuple, and uses that instead. How does that achieve what you want, especially since tuples are immutable? And this would require writing `.record([('value1', 'value2', 'etc')])`, if I understand correctly?